pkg/webhooks/scc/scc.go

```diff
@@ -0,0 +1,212 @@
+				Resources:   []string{"securitycontextconstraints"},
+				Scope:       &scope,
+			},
+		},
+	}
+	allowedUsers  = []string{"system:serviceaccount:openshift-monitoring:node-exporter"}
+	allowedGroups = []string{}
+	defaultSCCs   = []string{
+		"anyuid",
+		"hostaccess",
+		"hostmount-anyuid",
+		"hostnetwork",
+		"node-exporter",
+		"nonroot",
+		"privileged",
+		"restricted",
+		"pipelines-scc",
+	}
+)
+
+type SCCWebHook struct {
+	s runtime.Scheme
+}
+
+// NewWebhook creates the new webhook
+func NewWebhook() *SCCWebHook {
+	scheme := runtime.NewScheme()
+	admissionv1.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+
+	return &SCCWebHook{
+		s: *scheme,
+	}
+}
+
+// Authorized implements Webhook interface
+func (s *SCCWebHook) Authorized(request admissionctl.Request) admissionctl.Response {
+	return s.authorized(request)
+}
+
+func (s *SCCWebHook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
+	scc, err := s.renderSCC(request)
+	if err != nil {
+		log.Error(err, "Couldn't render a SCC from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	switch request.Operation {
+	case admissionv1.Delete:
+		if isDefaultSCC(scc) && !isAllowedUserGroup(request) {
+			log.Info(fmt.Sprintf("Deleting operation detected on default SCC: %v", scc.Name))
+			ret = admissionctl.Denied(fmt.Sprintf("Deleting default SCCs %v is not allowed", defaultSCCs))
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+	case admissionv1.Update:
+		if isDefaultSCC(scc) && !isAllowedUserGroup(request) {
+			log.Info(fmt.Sprintf("Updating operation detected on default SCC: %v", scc.Name))
+			ret = admissionctl.Denied(fmt.Sprintf("Modifying default SCCs %v is not allowed", defaultSCCs))
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+	}
+
+	ret = admissionctl.Allowed("Request is allowed")
+	ret.UID = request.AdmissionRequest.UID
+	return ret
+}
+
+// renderSCC render the SCC object from the requests
+func (s *SCCWebHook) renderSCC(request admissionctl.Request) (*securityv1.SecurityContextConstraints, error) {
+	decoder, err := admissionctl.NewDecoder(&s.s)
+	if err != nil {
+		return nil, err
+	}
+	scc := &securityv1.SecurityContextConstraints{}
+
+	if len(request.OldObject.Raw) > 0 {
+		err = decoder.DecodeRaw(request.OldObject, scc)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return scc, nil
+}
+
+// isAllowedUserGroup checks if the user or group is allowed to perform the action
+func isAllowedUserGroup(request admissionctl.Request) bool {
+	isAllowedUser := false
+	if len(allowedUsers) > 0 {
+		if utils.SliceContains(request.UserInfo.Username, allowedUsers) {
+			isAllowedUser = true
+		}
+	}
+
+	isAllowedGroup := false
+	if len(allowedGroups) > 0 {
+		for _, group := range allowedGroups {
+			if utils.SliceContains(group, request.UserInfo.Groups) {
+				isAllowedGroup = true
+			}
+		}
+	}
+
+	return (isAllowedUser || isAllowedGroup)
```

Comment:
I think this function could be simplified to something like
```suggestion
	if utils.SliceContains(request.UserInfo.Username, allowedUsers) {
		return true
	}

	for _, group := range allowedGroups {
		if utils.SliceContains(group, request.UserInfo.Groups) {
			return true
		}
	}
	return false
```